### DISCUSSION TITLE
Flutter Stream Exception in Realtime Client

### DISCUSSION THREAD

**Discussion by angus-clark** [supabase/supabase#5063] [Questions]
UPDATE: I now see there is an open issue in the supabase_flutter repo

I am currently making use of supabase's stream feature, however, some of the time I encounter an error that stops the stream from working as intended. I do not believe the error is directly related to my code but rather the package (not 100% sure though). The error is below:
`E/flutter (32644): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: type '_InternalLinkedHashMap<String, dynamic>' is not a subtype of type 'String?' in type cast
E/flutter (32644): #0      RealtimeSubscription.onError.<anonymous closure> (package:realtime_client/src/realtime_subscription.dart:97:42)
E/flutter (32644): #1      RealtimeSubscription.trigger (package:realtime_client/src/realtime_subscription.dart:225:20)
E/flutter (32644): #2      RealtimeClient.onConnMessage.<anonymous closure>.<anonymous closure> (package:realtime_client/src/realtime_client.dart:267:34)
E/flutter (32644): #3      Iterable.forEach (dart:core/iterable.dart:279:35)
E/flutter (32644): #4      RealtimeClient.onConnMessage.<anonymous closure> (package:realtime_client/src/realtime_client.dart:266:60)
E/flutter (32644): #5      new RealtimeClient.<anonymous closure> (package:realtime_client/src/realtime_client.dart:88:21)
E/flutter (32644): #6      RealtimeClient.onConnMessage (package:realtime_client/src/realtime_client.dart:251:11)
E/flutter (32644): #7      RealtimeClient.connect.<anonymous closure> (package:realtime_client/src/realtime_client.dart:110:11)
E/flutter (32644): #8      _rootRunUnary (dart:async/zone.dart:1436:47)
E/flutter (32644): #9      _CustomZone.runUnary (dart:async/zone.dart:1335:19)
E/flutter (32644): #10     _CustomZone.runUnaryGuarded (dart:async/zone.dart:1244:7)
E/flutter (32644): #11     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:341:11)
E/flutter (32644): #12     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
E/flutter (32644): #13     _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:123:11)
E/flutter (32644): #14     _HandleErrorStream._handleData (dart:async/stream_pipe.dart:253:10)
E/flutter (32644): #15     _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:153:13)
E/flutter (32644): #16     _rootRunUnary (dart:async/zone.dart:1436:47)
E/flutter (32644): #17     _CustomZone.runUnary (dart:async/zone.dart:1335:19)
E/flutter (32644): #18     _CustomZone.runUnaryGuarded (dart:async/zone.dart:1244:7)
E/flutter (32644): #19     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:341:11)
E/flutter (32644): #20     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
E/flutter (32644): #21     _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:733:19)
E/flutter (32644): #22     _StreamController._add (dart:async/stream_controller.dart:607:7)
E/flutter (32644): #23     _rootRunUnary (dart:async/zone.dart:1436:47)
E/flutter (32644): #24     _CustomZone.runUnary (dart:async/zone.dart:1335:19)
E/flutter (32644): #25     _CustomZone.runUnaryGuarded (dart:async/zone.dart:1244:7)
E/flutter (32644): #26     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:341:11)
E/flutter (32644): #27     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
E/flutter (32644): #28     _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:733:19)
E/flutter (32644): #29     _StreamController._add (dart:async/stream_controller.dart:607:7)
E/flutter (32644): #30     _StreamController.add (dart:async/stream_controller.dart:554:5)
E/flutter (32644): #31     new _WebSocketImpl._fromSocket.<anonymous closure> (dart:_http/websocket_impl.dart:1144:21)
E/flutter (32644): #32     _rootRunUnary (dart:async/zone.dart:1436:47)
E/flutter (32644): #33     _CustomZone.runUnary (dart:async/zone.dart:1335:19)
E/flutter (32644): #34     _CustomZone.runUnaryGuarded (dart:async/zone.dart:1244:7)
E/flutter (32644): #35     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:341:11)
E/flutter (32644): #36     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
E/flutter (32644): #37     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:63:11)
E/flutter (32644): #38     _EventSinkWrapper.add (dart:async/stream_transformers.dart:13:11)
E/flutter (32644): #39     _WebSocketProtocolTransformer._messageFrameEnd (dart:_http/websocket_impl.dart:332:23)
E/flutter (32644): #40     _WebSocketProtocolTransformer.add (dart:_http/websocket_impl.dart:226:46)
E/flutter (32644): #41     _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:111:24)
E/flutter (32644): #42     _rootRunUnary (dart:async/zone.dart:1436:47)
E/flutter (32644): #43     _CustomZone.runUnary (dart:async/zone.dart:1335:19)
E/flutter (32644): #44     _CustomZone.runUnaryGuarded (dart:async/zone.dart:1244:7)
E/flutter (32644): #45     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:341:11)
E/flutter (32644): #46     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
E/flutter (32644): #47     _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:733:19)
E/flutter (32644): #48     _StreamController._add (dart:async/stream_controller.dart:607:7)
E/flutter (32644): #49     _StreamController.add (dart:async/stream_controller.dart:554:5)
E/flutter (32644): #50     _Socket._onData (dart:io-patch/socket_patch.dart:2302:41)
E/flutter (32644): #51     _rootRunUnary (dart:async/zone.dart:1436:47)
E/flutter (32644): #52     _CustomZone.runUnary (dart:async/zone.dart:1335:19)
E/flutter (32644): #53     _CustomZone.runUnaryGuarded (dart:async/zone.dart:1244:7)
E/flutter (32644): #54     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:341:11)
E/flutter (32644): #55     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
E/flutter (32644): #56     _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:733:19)
E/flutter (32644): #57     _StreamController._add (dart:async/stream_controller.dart:607:7)
E/flutter (32644): #58     _StreamController.add (dart:async/stream_controller.dart:554:5)
E/flutter (32644): #59     _RawSecureSocket._sendReadEvent (dart:io/secure_socket.dart:1000:19)
E/flutter (32644): #60     _rootRun (dart:async/zone.dart:1420:47)
E/flutter (32644): #61     _CustomZone.run (dart:async/zone.dart:1328:19)
E/flutter (32644): #62     _CustomZone.runGuarded (dart:async/zone.dart:1236:7)
E/flutter (32644): #63     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1276:23)
E/flutter (32644): #64     _rootRun (dart:async/zone.dart:1428:13)
E/flutter (32644): #65     _CustomZone.run (dart:async/zone.dart:1328:19)
E/flutter (32644): #66     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:1260:23)
E/flutter (32644): #67     Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:18:15)
E/flutter (32644): #68     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:395:19)
E/flutter (32644): #69     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:426:5)
E/flutter (32644): #70     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:192:12)
E/flutter (32644): `

Any help would be greatly appreciated.

**Comment by w3b6x9** [comment]
there's a new `supabase-dart` version that addresses this. see [comment](https://github.com/supabase-community/realtime-dart/issues/35#issuecomment-1025008931) for more details.
